lib/registration.pm

```diff
@@ -274,7 +274,7 @@ sub register_addons {
         # WE doesn't need code on SLED
         push @addons_with_code, 'we' unless (check_var('SLE_PRODUCT', 'sled'));
         # HA doesn't need code on SLES4SAP
-        push @addons_with_code, 'ha' unless (check_var('SLE_PRODUCT', 'sles4sap') || (is_upgrade && is_sle('12-sp5+')));
+        push @addons_with_code, 'ha' unless (check_var('SLE_PRODUCT', 'sles4sap') || (is_upgrade && is_sle('12-sp5+') && !is_online_migration));
```

Comment:
`get_var('UPGRADE') || get_var('ONLINE_MIGRATION')  || get_var('ZDUP') || get_var('AUTOUPGRADE') || get_var('LIVE_UPGRADE') && ! get_var('ONLINE_MIGRATION') `  - so if you replace `is_upgrade` call with it content you will get this . So looks like adding `is_online_migration` inside `is_upgrade` does not make sense . Can you show me place where it really makes . In anycase the least what you can do is put comment here explaining why you putting contradicting conditions into the same logical expression .
Side note : yeah I see that it will work but it may confuse people ...